vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -495,17 +498,30 @@ def collect_events(self):
             self.log.debug("Got %s new events from the vCenter event manager", len(new_events))
             event_config = {'collect_vcenter_alarms': True}
             for event in new_events:
+                self.log.debug(
+                    "Processing event number:%s, type:%s: msg:%s", event.key, type(event), event.fullFormattedMessage
+                )
                 normalized_event = VSphereEvent(event, event_config, self.config.base_tags)
                 # Can return None if the event if filtered out
                 event_payload = normalized_event.get_datadog_payload()
                 if event_payload is not None:
+                    self.log.debug(
+                        "Submit event number:%s, type:%s: msg:%s", event.key, type(event), event.fullFormattedMessage
+                    )
                     self.event(event_payload)
+                if latest_event_time is None or event.createdTime > latest_event_time:
+                    latest_event_time = event.createdTime
         except Exception as e:
             # Don't get stuck on a failure to fetch an event
             # Ignore them for next pass
             self.log.warning("Unable to fetch Events %s", e)
 
-        self.latest_event_query = self.api.get_latest_event_timestamp() + dt.timedelta(seconds=1)
+        if latest_event_time is not None:
+            self.latest_event_query = latest_event_time + dt.timedelta(seconds=1)
+        else:
+            # Let's set `self.latest_event_query` to `collect_start_time` as safeguard in case no events are reported
+            # OR something bad happened (which might happen again indefinitely).
+            self.latest_event_query = collect_start_time
```

Comment:
We'll end up in a loop here on the first unparseable event. Which was not the case before as we were skipping events with errors (dropping the whole batch).

Reply:
I don't think we are in a loop, except if I'm missing something.

If there is a failure because of a unparseable event, `self.latest_event_query` is set to `collect_start_time` and the batch is skipped.

At the moment (master branch), we have a loop I think, when `self.latest_event_query = self.api.get_latest_event_timestamp() + dt.timedelta(seconds=1)` fails when the `get_latest_event_timestamp` can't be retrieved. The way out, is that the last event become valid again.

Reply:
Oh my bad I misunderstood the part where collect_start_time gets updated every time.
In that case that seems correct and we don't end up in a loop 👍